src/ape/api/query.py

```diff
@@ -0,0 +1,108 @@
+import sys
+from typing import Any, Dict, List, Optional, Union
+
+import pandas as pd
+from ethpm_types.abi import EventABI, MethodABI
+from pydantic import BaseModel, PositiveInt, root_validator
+
+from ape.managers.networks import NetworkManager
+from ape.types import AddressType
+from ape.utils import abstractdataclass, abstractmethod
+
+# We can remove this once we stop supporting python3.7.
+if sys.version_info >= (3, 8):
```

Comment:
We should create a `_compat.py` file in the `ape` folder and import `Literal` from there both here and `ape.types` and where-ever else it's used. That way this mess is only in one spot.

Reply:
Example: https://github.com/commaai/usr/blob/master/local/lib/python2.7/site-packages/service_identity/_compat.py